src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,249 @@
+            statement.setLong(FIRST_INDEX, previousCarVersion.getId());
+            for (Driver driver : previousCarVersion.getDrivers()) {
+                statement.setLong(SECOND_INDEX, driver.getId());
+                statement.executeUpdate();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Couldn't delete "
+                    + "relations between " + previousCarVersion
+                    + " and drivers.", e);
+        }
+    }
+
+    private Car addDriverToCar(ResultSet resultSet, Car foundedCar) throws SQLException {
+        Driver newDriver = initializeDriver(resultSet);
+        List<Driver> drivers = new ArrayList<>(List.copyOf(foundedCar.getDrivers()));
+        drivers.add(newDriver);
+        foundedCar.setDrivers(drivers);
+        return foundedCar;
+    }
+
+    private Car initializedNewCar(ResultSet resultSet) throws SQLException {
+        Long carId = resultSet.getObject("car_id", Long.class);
+        String model = resultSet.getString("model");
+        Manufacturer manufacturer = initializeManufacturer(resultSet);
+        Driver driver = initializeDriver(resultSet);
+        List<Driver> drivers = List.of(driver);
+        return new Car(carId, model, manufacturer, drivers);
+    }
+
+    private Manufacturer initializeManufacturer(ResultSet resultSet) throws SQLException {
+        Long manufacturerId = resultSet.getObject("manufacturer_id", Long.class);
+        String manufacturerName = resultSet.getString("manufacturer_name");
+        String country = resultSet.getString("country");
+        return new Manufacturer(manufacturerId, manufacturerName, country);
+    }
+
+    private Driver initializeDriver(ResultSet resultSet) throws SQLException {
+        Long driverId = resultSet.getObject("driver_id", Long.class);
+        String driverName = resultSet.getString("driver_name");
+        String licenseNumber = resultSet.getString("license_number");
+        return new Driver(driverId, driverName, licenseNumber);
+    }
+
+    private void setRelationsCarDrivers(Car car) {
+        if (car.getDrivers().isEmpty()) {
+            return;
+        }
+        String insertRequest = "INSERT INTO cars_drivers"
+                               + " (cars_id, drivers_id) VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(insertRequest,
+                         Statement.RETURN_GENERATED_KEYS)) {
+            statement.setLong(FIRST_INDEX, car.getId());
+            for (Driver driver : car.getDrivers()) {
+                statement.setLong(SECOND_INDEX, driver.getId());
+                statement.executeUpdate();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't create "
+                                              + car + ". ", e);
```

Comment:
Do you creatin a car, or relation between car and drivers? Let's change the error message and and more details to it (carId and driverId).